save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt

```diff
@@ -49,24 +63,43 @@ class TestExecutionController(private val testExecutionService: TestExecutionSer
      * @param status
      * @param testSuite
      * @param authentication
+     * @param checkDebugInfo if true, response will contain information about whether debug info data is available for this test execution
      * @return a list of [TestExecutionDto]s
      */
     @GetMapping(path = ["/api/$v1/testExecutions"])
+    @Parameter(
+        `in` = ParameterIn.HEADER,
+        name = "X-Authorization-Source",
+        required = true,
+        example = "basic"
+    )
     @Suppress("LongParameterList", "TOO_MANY_PARAMETERS", "TYPE_ALIAS")
     fun getTestExecutions(
         @RequestParam executionId: Long,
         @RequestParam page: Int,
         @RequestParam size: Int,
         @RequestParam(required = false) status: TestResultStatus?,
         @RequestParam(required = false) testSuite: String?,
+        @RequestParam(required = false, defaultValue = "false") checkDebugInfo: Boolean,
```

Comment:
Why defaultValue = "false"?

Reply:
We should perform additional operations (here - some filesystem queries) only if it's explicitly requested by the consumer

Reply:
But if it's not requested, but debug info doesn't exist. we will get previous problem again?

Reply:
It should be handled on frontend properly: if there is a possibility that debug info is requested, then its presence should be checked. If component doesn't show any related info, then it shouldn't care whether it's available or not.
Also - I've updated the PR and made this field nullable. This way we can distinguish cases when debug info is unavailable and when it hassn't been checked

Reply:
Didn't look, that this method used not only by frontend
Sure, of course, should be `false`

Reply:
It's used currently only by frontend, but in several places, only some of them are related to debug info. If this method is used by API integrations, then they also may need this information only someties